### PR TITLE
Users-Add-Media

### PR DESCRIPTION
**Pull request by MarioGafeed**
Add Media Library To Users
Plz Confirm me to Continue with all Models

### CODE REVIEW

File: app/Http/Repositories/UserRepository.php [hunk 1]
```diff
@@ -124,9 +131,9 @@ public function destroy($id)
     {
       $redirect = true;
       $user = $this->getById($id );
-      // if (file_exists(public_path('uploads/' . $user->image))) {
-      //     @unlink(public_path('uploads/' . $user->image));
-      // }
+      if (file_exists(public_path($user->getFirstMedia()->getUrl()))) {
```
Comment (mo7zayed) [inline app/Http/Repositories/UserRepository.php:134]: Don't use this, Don't check the file existence, `file_exists` checks the file path on local disk, Just use clear clearMediaCollection() and it will delete the file by default

File: config/filesystems.php [hunk 2]
```diff
@@ -42,6 +42,12 @@
             'visibility' => 'public',
         ],
 
+        'media' => [
```
Comment (mo7zayed) [inline config/filesystems.php:45]: why did you add a custom driver for the media? you don't need to do that you just use the default driver

File: config/media-library.php [hunk 3]
```diff
@@ -0,0 +1,223 @@
+<?php
+
+return [
+
+    /*
+     * The disk on which to store added files and derived images by default. Choose
+     * one or more of the disks you've configured in config/filesystems.php.
+     */
+    // 'disk_name' => env('MEDIA_DISK', 'public'),
```
Comment (mo7zayed) [inline config/media-library.php:9]: just use the default disk for it

Comment (mo7zayed) [reply config/media-library.php:9]: which is `public`

File: resources/views/backend/users/incs/_fields.blade.php [hunk 4]
```diff
@@ -11,6 +11,32 @@
         </div>
     </div>
 
+    <div class="form-group{{ $errors->has('image') ? ' has-error' : '' }}">
+       <label class="control-label col-md-2">{{ trans('main.image') }}</label>
+       <div class="col-md-10">
+           <div class="fileinput fileinput-new" data-provides="fileinput">
+               <div class="fileinput-preview thumbnail" data-trigger="fileinput" style="width: 200px; height: 150px;">
+                   @if (checkValue(getData($data, 'image')))
+                       <img src="{{ ShowImage(getData($data, 'image')) }}" alt="" />
```
Comment (mo7zayed) [inline resources/views/backend/users/incs/_fields.blade.php:20]: You should use `$data->getFirstMedia()->getUrl()` incase editing the form

File: config/media-library.php (hunk 3 above)
Comment (MarioGafeed) [reply config/media-library.php:9]: The Default Disk is::
 'public' => [
            'driver' => 'local',
            'root' => storage_path('app/public'),
            'url' => env('APP_URL').'/storage',
            'visibility' => 'public',
        ],
+++++++++++++++++++++++++++++++++++++++++
مش المفروض اسجل الداتا في الــ PUBLIC??

File: config/filesystems.php (hunk 2 above)
Comment (MarioGafeed) [reply config/filesystems.php:45]: Done
